### PR TITLE
Loadbalancer resources

### PR DESCRIPTION
**Pull request by Overglazed**

### CODE REVIEW

File: ecloud/data_source_loadbalancer_network.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+	if len(lbNetworks) > 1 {
+		return errors.New("More than 1 loadbalancer network found with provided arguments")
+	}
+
+	d.SetId(lbNetworks[0].ID)
+	d.Set("name", lbNetworks[0].Name)
+	d.Set("config_id", lbNetworks[0].NetworkID)
+	d.Set("load_balancer_id", lbNetworks[0].LoadBalancerID)
+
```
Comment (0x4c6565) [inline ecloud/data_source_loadbalancer_network.go:72]: Need to populate `network_id` here

File: docs/data-sources/ecloud_loadbalancer_network.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+- `loadbalancer_id`: ID of loadbalancer
+- `network_id`: ID of ecloud network
+
+## Attributes Reference
+
+`id` is set to loadbalancer network ID
+
+- `name`: Name of loadbalancer network
+- `network_id`: ID of ecloud network
+- `loadbalancer_id`: ID of loadbalancer
```
Comment (0x4c6565) [inline docs/data-sources/ecloud_loadbalancer_network.md:27]: Missing `network_id` here

File: ecloud/resource_loadbalancer.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+				Computed: true,
+			},
+			"load_balancer_spec_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"availability_zone_id": {
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (0x4c6565) [inline ecloud/resource_loadbalancer.go:44]: Could do with `ForceNew: true` for this property

File: ecloud/resource_loadbalancer.go [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+		Update: resourceLoadBalancerUpdate,
+		Delete: resourceLoadBalancerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (0x4c6565) [inline ecloud/resource_loadbalancer.go:26]: Could do with `ForceNew: true` for this property

File: ecloud/resource_loadbalancer_network.go [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"load_balancer_id": {
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (0x4c6565) [inline ecloud/resource_loadbalancer_network.go:34]: Could do with `ForceNew: true` for this property

File: ecloud/resource_loadbalancer_network.go [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+		Update: resourceLoadBalancerNetworkUpdate,
+		Delete: resourceLoadBalancerNetworkDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"network_id": {
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (0x4c6565) [inline ecloud/resource_loadbalancer_network.go:26]: Could do with `ForceNew: true` for this property

**Review by 0x4c6565 — CHANGES_REQUESTED** [review]
Just a few comments :)

File: ecloud/data_source_loadbalancer_network.go (hunk 1 above)
Comment (Overglazed) [reply ecloud/data_source_loadbalancer_network.go:73]: oops missed that!

File: ecloud/resource_loadbalancer_network.go (hunk 5 above)
Comment (Overglazed) [reply ecloud/resource_loadbalancer_network.go:35]: added

File: ecloud/resource_loadbalancer_network.go (hunk 6 above)
Comment (Overglazed) [reply ecloud/resource_loadbalancer_network.go:26]: added

**Review by Overglazed — COMMENTED** [review]
resolving pr comments

**Comment by Overglazed** [comment]
Would help if I added forceNew to the correct resource! 

Should be good now to rereview @0x4c6565  :)

File: ecloud/resource_loadbalancer_vip.go [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+		return fmt.Errorf("Error waiting for loadbalancer vip with ID [%s] to be deleted: %s", d.Id(), err)
+	}
+
+	//remove floating ip if set
+	if len(d.Get("floating_ip_id").(string)) > 1 {
+		fip := d.Get("floating_ip_id").(string)
+
+		log.Printf("[DEBUG] Unassigning floating ip with ID [%s]", fip)
+
+		taskID, err := service.UnassignFloatingIP(fip)
```
Comment (0x4c6565) [inline ecloud/resource_loadbalancer_vip.go:186]: I don't believe unassign is required here - removing the VIP will unassign the floating IP from the VIP ip address if i'm not mistaken?

**Review by 0x4c6565 — APPROVED** [review]
LGTM, just dropped a single comment

File: ecloud/resource_loadbalancer_vip.go (hunk 7 above)
Comment (Overglazed) [reply ecloud/resource_loadbalancer_vip.go:186]: Good catch. Yeah that was from before I switched the delete ordering. Will get that cleaned up

**Pull request merged**